section_to_course/api/views.py

```diff
@@ -0,0 +1,68 @@
+"""
+Helper API endpoints for the section to course application.
+"""
+
+from django.utils.translation import gettext as _
+from opaque_keys import InvalidKeyError
+from opaque_keys.edx.keys import CourseKey
+from rest_framework import status
+from rest_framework.permissions import IsAdminUser
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from xmodule.modulestore.django import modulestore
+
+
+class CourseAutocomplete(APIView):
+    """
+    Autocomplete API endpoint for courses.
+    """
+
+    permission_classes = [IsAdminUser]
+
+    def get(self, request):
+        """
+        Get all courses and match a search term against them.
+        """
+        self.check_permissions(request)
+        # TODO: More efficient way of doing this. Do we have access to a search index?
+        all_courses = [{'id': str(course.id), 'text': course.display_name} for course in modulestore().get_courses()]
```

Comment:
I tried looking for an API function that would meet requirements here, but I did not see one. I **could** query the LearningSequence models directly if desired-- or add an additional upstream API function if it's needed. Of the two, I think the latter is a better option, however I'd suggest if we do that, we do it in a follow-up, and keep the implementation as it is now. What say you?

Reply:
Which APIs are you missing? We can use [this one](https://github.com/openedx/edx-platform/blob/299cd981e5594e429aa464c7eef7d9c5db9913ad/openedx/core/djangoapps/content/learning_sequences/api/outlines.py#L87-L96) to get all courses and [this one](https://github.com/openedx/edx-platform/blob/299cd981e5594e429aa464c7eef7d9c5db9913ad/openedx/core/djangoapps/content/learning_sequences/api/outlines.py#L99-L194) to retrieve sections (something like `get_course_outline(course_id).sections` should work, according to [this data class](https://github.com/openedx/edx-platform/blob/a131d6360863d3e168ab303ff0a9b878ce6e6439/openedx/core/djangoapps/content/learning_sequences/data.py#L204)).

Reply:
The first one (get_course_keys_with_outline) doesn't give you display names, which means I'd have to make separate queries for each entry, so it's not much of an improvement performance-wise. It's also not letting me filter by display-name, so I'm still iterating over them all. Not ideal.

I'm currently trying out the second one. It might work, but the LearningSequence models aren't being automatically created as I expected for the courses I'm creating in my tests. I'll update when I have more info.

Reply:
Looks like it was just that the factories don't trigger the course published signal. So I added a manual call to it.

Reply:
@Kelketek, do we want to use display names instead of course keys? If you look at the client's Studio, you will see many courses with identical names, so querying by course keys could be more efficient for course authors.

Reply:
@Agrendalath If they input the course key, it will still narrow down to only one option in the autoselect. However to give a bit more certainty to the user I've added the course key to the display in the autoselect.